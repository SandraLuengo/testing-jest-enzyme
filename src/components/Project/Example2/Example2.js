import React from 'react';
import { connect } from 'react-redux';
import { simpleAction } from '../../../actions/simpleAction';

const mapStateToProps = state => ({
  contact: state.simpleReducer.contact,

});

const mapDispatchToProps = dispatch => ({
  simpleAction: (variable) => dispatch(simpleAction(variable)),
});

const Example2 = ({ className, simpleAction, contact }) => {
  const handleChange = ({ target: { value } }) => {
    simpleAction(value);
  };
  return (<div className={`${className} project__example2`}>
    <h2>Example 2: search</h2>
    <input className="searchInput" type="text" onChange={e => handleChange(e)} />
    <button className="project__example2__send" >Send</button>
    <p className="text" >{contact}</p>
  </div>);
};

export default connect(mapStateToProps, mapDispatchToProps)(Example2);

