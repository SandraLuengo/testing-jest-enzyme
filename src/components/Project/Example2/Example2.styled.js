import styled from 'styled-components';
import Example2 from './Example2';

export default styled(Example2).attrs({})``;
