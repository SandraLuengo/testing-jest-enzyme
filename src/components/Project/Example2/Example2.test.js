import React from 'react';
import { Provider } from 'react-redux';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Example2 from './Example2';
import configureStore from '../../../store';

const initialState = {
  counter: '',
};
const store = configureStore(initialState);

describe('<Example2/>', () => {
  const wrapper = mount(<Provider store={configureStore()}><Example2 /></Provider>);
  it('renders correctly', () => {
    shallow(<Provider store={configureStore()}><Example2 /></Provider>);
    const wrapper2 = shallow(<Provider store={configureStore()}><Example2 /></Provider>);
    const component = wrapper2.dive();
    expect(toJson(component)).toMatchSnapshot();
  });
  it('it should exist an input type text', () => {
    expect(wrapper.find('input[type="text"]').length).toBe(1);
  });
  it('should change search value', () => {
    wrapper
      .find('.searchInput')
      .simulate('change', { target: { value: 'testing' } });
    expect(wrapper.find('.text').text()).toBe('testing');
  });
  it('dispatch', () => {
    const dispatch = store.dispatch({
      type: 'SIMPLE_ACTION',
      newValue: 'pepe',
    });
    expect(dispatch).toMatchSnapshot();
  });
});
