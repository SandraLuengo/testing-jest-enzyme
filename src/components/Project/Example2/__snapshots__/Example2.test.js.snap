// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Example2/> dispatch 1`] = `
Object {
  "newValue": "pepe",
  "type": "SIMPLE_ACTION",
}
`;

exports[`<Example2/> renders correctly 1`] = `<Connect(Example2) />`;
